packages/flutter_tools/lib/src/base/bot_detector.dart

```diff
@@ -5,56 +5,108 @@
     }
 
-    return _platform.environment['BOT'] == 'true'
+    return _isRunningOnBot = _platform.environment['BOT'] == 'true'
 
-        // https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables
-        || _platform.environment['TRAVIS'] == 'true'
-        || _platform.environment['CONTINUOUS_INTEGRATION'] == 'true'
-        || _platform.environment.containsKey('CI') // Travis and AppVeyor
+      // https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables
+      || _platform.environment['TRAVIS'] == 'true'
+      || _platform.environment['CONTINUOUS_INTEGRATION'] == 'true'
+      || _platform.environment.containsKey('CI') // Travis and AppVeyor
 
-        // https://www.appveyor.com/docs/environment-variables/
-        || _platform.environment.containsKey('APPVEYOR')
+      // https://www.appveyor.com/docs/environment-variables/
+      || _platform.environment.containsKey('APPVEYOR')
 
-        // https://cirrus-ci.org/guide/writing-tasks/#environment-variables
-        || _platform.environment.containsKey('CIRRUS_CI')
+      // https://cirrus-ci.org/guide/writing-tasks/#environment-variables
+      || _platform.environment.containsKey('CIRRUS_CI')
 
-        // https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
-        || (_platform.environment.containsKey('AWS_REGION') &&
-            _platform.environment.containsKey('CODEBUILD_INITIATOR'))
+      // https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
+      || (_platform.environment.containsKey('AWS_REGION') &&
+          _platform.environment.containsKey('CODEBUILD_INITIATOR'))
 
-        // https://wiki.jenkins.io/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-belowJenkinsSetEnvironmentVariables
-        || _platform.environment.containsKey('JENKINS_URL')
+      // https://wiki.jenkins.io/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-belowJenkinsSetEnvironmentVariables
+      || _platform.environment.containsKey('JENKINS_URL')
 
-        // Properties on Flutter's Chrome Infra bots.
-        || _platform.environment['CHROME_HEADLESS'] == '1'
-        || _platform.environment.containsKey('BUILDBOT_BUILDERNAME')
-        || _platform.environment.containsKey('SWARMING_TASK_ID');
+      // Properties on Flutter's Chrome Infra bots.
+      || _platform.environment['CHROME_HEADLESS'] == '1'
+      || _platform.environment.containsKey('BUILDBOT_BUILDERNAME')
+      || _platform.environment.containsKey('SWARMING_TASK_ID')
+      || await _azureDetector.isRunningOnAzure;
   }
 }
 
-bool isRunningOnBot(Platform platform) {
-  final BotDetector botDetector = context.get<BotDetector>() ?? BotDetector(platform: platform);
-  return botDetector.isRunningOnBot;
+// Are we running on Azure?
+// https://docs.microsoft.com/en-us/azure/virtual-machines/linux/instance-metadata-service
+class _AzureDetector {
```

Comment:
Make this public and `@visibleForTesting` so you can unit test it?

Reply:
Ahh, right. Done.